plugin-server/src/worker/vm/extensions/posthog.ts

```diff
@@ -27,21 +28,35 @@ async function queueEvent(hub: Hub, pluginConfig: PluginConfig, data: InternalDa
     partitionKeyHash.update(`${data.team_id}:${data.distinct_id}`)
     const partitionKey = partitionKeyHash.digest('hex')
 
+    const message = JSON.stringify({
+        distinct_id: data.distinct_id,
+        ip: '',
+        site_url: '',
+        data: JSON.stringify(data),
+        team_id: pluginConfig.team_id,
+        now: data.timestamp,
+        sent_at: data.timestamp,
+        uuid: data.uuid,
+    } as RawEventMessage)
+
+    // We currently don't drop or truncate oversized events, although Kafka will reject the messages anyway.
+    // Log these messages so that we want dig into what even type triggers KafkaJSProtocolError when we see them.
+    const messageSize = Buffer.from(message).length
+    if (messageSize > 1_000_000) {
```

Comment:
Non-blocking flyby: this is duplicating logic within plugin-server/src/utils/db/kafka-producer-wrapper.ts - See estimateMessageSize.

Reply:
Agreed, it's lifted from that function.
As I'm not yet confident about the perf impact of my changes I try to be conservative and not stringify the whole kafka record one more time.

Would you have called `estimateMessageSize(record)` here? I want to do it before calling `queueMessage` so that I still have the plugin name context.

Reply:
As long as we're aware of the duplication I trust you and consider this non-blocking :)